### PR TITLE
Avoid python or R install during most docker builds

### PR DESCRIPTION
**Pull request by TedBrookings**
There are now two parallel lines of docker images:
1) Python, R, and other fixed software dependencies are mostly built in
   a series of dockers with names ending in "-virtual-env". These
   dockers should only need to be rebuilt when we specifically need
   and want to change our 3rd-party software dependencies.
2) Depedencies on our own code live in the same dockers as before, but
   have their critical 3rd-party software dependencies copied in from
   the corresponding "-virtual-env" image.

### CODE REVIEW

**Comment by TedBrookings** [comment]
- I rebuilt all the dockers dependent on sv-base-mini, and the updated images are of course listed in dockers.json.
- I ran MakeCohortVcf and it succeeded. The workflow ID is `cb9f1829-cfed-4362-8bad-cab5bb155871` and the output folder is `gs://broad-methods-cromwell-exec-bucket-instance-8/MakeCohortVcf/cb9f1829-cfed-4362-8bad-cab5bb155871`
- I also ran GatherBatchEvidence but I've had trouble with out-of-memory errors with gcnv. I doubled the memory and the current workflow is `d89c3e06-7048-49c7-8f95-2c527653e0ff` and output bucket is `broad-methods-cromwell-exec-bucket-instance-8/GatherBatchEvidence/d89c3e06-7048-49c7-8f95-2c527653e0ff`. cnmops, which was the only docker not tested by MakeCohortVcf has completed, but some gcnv tasks are stuck at "Starting".  In the interest of getting this reviewed before more changes are made to the dockers and the merge gets _challenging_, I made the PR now.

**Comment by TedBrookings** [comment]
One little bit more of summary info. The advantages of this approach vs the old one:

- Python, R, and other software dependencies will be absolutely pinned, both because the new build approach fixes them a bit more, but also because those dockers should only rarely be re-built, and thus cannot change.

File: dockerfiles/sv-pipeline-virtual-env/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+RUN install_bioconductor_packages.R $SV_PIPELINE_BIOCONDUCTOR_PKGS
+
+# Install plink2 & KING (for relatedness inference)
+ARG KING_URL="https://www.kingrelatedness.com/executables/Linux-king215.tar.gz"
+RUN cd $VIRTUAL_ENV_PATH && \
+    wget -q $KING_URL && \
+    tar -xzf Linux-king215.tar.gz && \
+    rm -f Linux-king215.tar.gz
+
+ARG PLING2_URL="https://github.com/chrchang/plink-ng/releases/download/2019/plink2_linux_x86_64_20190107.zip"
```
Comment (mwalker174) [inline dockerfiles/sv-pipeline-virtual-env/Dockerfile:22]:
```suggestion
ARG PLINK2_URL="https://github.com/chrchang/plink-ng/releases/download/2019/plink2_linux_x86_64_20190107.zip"
```

**Review by mwalker174 — APPROVED** [review]
Thank you @TedBrookings this is a big QOL improvement for development! As we discussed offline, the number of docker images is getting a bit high so we should definitely look at addressing #261 next. I tested a full build and got these sizes:
```
us.gcr.io/broad-dsde-methods/markw/sv-pipeline-qc                     tb-fix-dockers-r-build-de17778                      2d9f1b53824c        19 hours ago        3.17GB
us.gcr.io/broad-dsde-methods/markw/sv-pipeline-rdtest                 tb-fix-dockers-r-build-de17778                      927520b9b5f0        19 hours ago        3.21GB
us.gcr.io/broad-dsde-methods/markw/sv-pipeline-rdtest-virtual-env     tb-fix-dockers-r-build-de17778                      498d2133880e        19 hours ago        8.1GB
us.gcr.io/broad-dsde-methods/markw/sv-pipeline-hail                   tb-fix-dockers-r-build-de17778                      c6854ecaf4bc        19 hours ago        5.16GB
us.gcr.io/broad-dsde-methods/markw/sv-pipeline-qc-virtual-env         tb-fix-dockers-r-build-de17778                      357b93dd47d3        19 hours ago        8.06GB
us.gcr.io/broad-dsde-methods/markw/sv-pipeline-updates                tb-fix-dockers-r-build-de17778                      f626a7cd3613        20 hours ago        4.61GB
us.gcr.io/broad-dsde-methods/markw/sv-pipeline-children-virtual-env   tb-fix-dockers-r-build-de17778                      f102dd3cc54d        20 hours ago        6.12GB
us.gcr.io/broad-dsde-methods/markw/sv-pipeline                        tb-fix-dockers-r-build-de17778                      da74caff86a4        20 hours ago        2.93GB
us.gcr.io/broad-dsde-methods/markw/sv-pipeline-hail-virtual-env       tb-fix-dockers-r-build-de17778                      d64ea7ed69d3        20 hours ago        4.91GB
us.gcr.io/broad-dsde-methods/markw/sv-pipeline-updates-virtual-env    tb-fix-dockers-r-build-de17778                      39e79a3685ae        20 hours ago        6GB
us.gcr.io/broad-dsde-methods/markw/sv-pipeline-base                   tb-fix-dockers-r-build-de17778                      fe1a0950afb8        20 hours ago        2.85GB
us.gcr.io/broad-dsde-methods/markw/sv-pipeline-virtual-env            tb-fix-dockers-r-build-de17778                      e2f113a640a5        20 hours ago        4.28GB
us.gcr.io/broad-dsde-methods/markw/cnmops                             tb-fix-dockers-r-build-de17778                      7251326fd570        20 hours ago        1.76GB
us.gcr.io/broad-dsde-methods/markw/sv-pipeline-base-virtual-env       tb-fix-dockers-r-build-de17778                      bc3fbdf51d1e        20 hours ago        2.6GB
us.gcr.io/broad-dsde-methods/markw/samtools-cloud                     tb-fix-dockers-r-build-de17778                      e7c6f8d06690        21 hours ago        451MB
us.gcr.io/broad-dsde-methods/markw/sv-base                            tb-fix-dockers-r-build-de17778                      7740689be00c        21 hours ago        1.53GB
us.gcr.io/broad-dsde-methods/markw/cnmops-virtual-env                 tb-fix-dockers-r-build-de17778                      6e5284943e32        21 hours ago        1.63GB
us.gcr.io/broad-dsde-methods/markw/sv-base-virtual-env                tb-fix-dockers-r-build-de17778                      643bc9bab364        21 hours ago        998MB
us.gcr.io/broad-dsde-methods/markw/manta                              tb-fix-dockers-r-build-de17778                      07d656c66887        21 hours ago        253MB
us.gcr.io/broad-dsde-methods/markw/delly                              tb-fix-dockers-r-build-de17778                      92815dd3368f        21 hours ago        14.2MB
us.gcr.io/broad-dsde-methods/markw/sv-base-mini                       tb-fix-dockers-r-build-de17778                      b0fde947975f        21 hours ago        184MB
us.gcr.io/broad-dsde-methods/markw/wham                               tb-fix-dockers-r-build-de17778                      48c0b6d77d20        21 hours ago        162MB
```
I noticed that some of the virtual-env images are quite large (>8GB). This surprising so I was wondering if there was a way to reduce their size?

**Comment by TedBrookings** [comment]
Those image sizes are large in part due to the fact that they are uncompressed (I assume they are what's listed on your local docker build). The compressed sizes are much smaller, and that's what matters because that is what is transmitted. That said, there is still obviously a problem because they are bigger than they should be. e.g. the compressed sv-pipeline-virtual-env is 2G, but sv-pipeline is only 1.1GB. That means that there must be a lot of junk in sv-pipeline-virtual-env that is not being copied over. I set up the build of the virtual env images to try to avoid accumulating junk outside the virtual env (that's what the last copy step in those dockerfiles is for), but obviously it didn't work as well as I hoped. That should be fixable. I'll address that when I go after issue #261 .

**Pull request merged**